packages/lwc/types.d.ts

```diff
@@ -178,4 +180,24 @@ declare module 'lwc' {
      * @param config configuration object for the accessor
      */
     export function wire(getType: (config?: any) => any, config?: any): PropertyDecorator;
+
+    type ConfigValue = Record<string, any>;
+    type ContextValue = Record<string, any>;
+
+    interface WireAdapter {
+        update(config: ConfigValue, context?: ContextValue): void;
+        connect(): void;
+        disconnect(): void;
+    }
+
+    type DataCallback = (value: any) => void;
+    type WireAdapterSchemaValue = 'optional' | 'required';
+
+    interface WireAdapterConstructor {
+        new (callback: DataCallback): WireAdapter;
+        configSchema?: Record<string, WireAdapterSchemaValue>;
+        contextSchema?: Record<string, WireAdapterSchemaValue>;
+    }
+
+    export function createContextProvider(config: WireAdapterConstructor): void;
```

Comment:
This one does have a return a value: check [here](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-core/src/framework/context-provider.ts#L36).